XenAdmin/Wizards/CrossPoolMigrateWizard/Filters/CrossPoolMigrateFilter.cs

```diff
@@ -90,109 +90,71 @@ protected override bool FailureFoundFor(IXenObject itemToFilterOn, out string fa
-                        var targetNetwork = GetANetwork(host);
+                    //CA-205799: do not offer the host the VM is currently on
+                    Host homeHost = vm.Home();
+                    if (homeHost != null && homeHost.opaque_ref == host.opaque_ref)
+                        continue;
 
-                        VM.assert_can_migrate(vm.Connection.Session,
-                            vm.opaque_ref,
-                            receiveMapping,
-                            true,
-                            GetVdiMap(vm, targetSrs),
-                            vm.Connection == host.Connection ? new Dictionary<XenRef<VIF>, XenRef<XenAPI.Network>>() : GetVifMap(vm, targetNetwork),
-                            new Dictionary<string, string>());
+                    //check if the destination host is older than the source host
+                    var destinationVersion = Helpers.HostPlatformVersion(host);
+                    var sourceVersion = Helpers.HostPlatformVersion(vm.Home() ?? Helpers.GetCoordinator(vmPool));
 
-                        break;
+                    if (Helpers.ProductVersionCompare(destinationVersion, sourceVersion) < 0)
+                    {
+                        failureReason = Messages.OLDER_THAN_CURRENT_SERVER;
+                        continue;
                     }
-                    catch (Failure failure)
+
+                    if (Host.RestrictDMC(host) &&
+                        (vm.power_state == vm_power_state.Running ||
+                         vm.power_state == vm_power_state.Paused ||
+                         vm.power_state == vm_power_state.Suspended) &&
+                        (vm.memory_static_min > vm.memory_dynamic_min || //corner case, probably unlikely
+                         vm.memory_dynamic_min != vm.memory_dynamic_max ||
+                         vm.memory_dynamic_max != vm.memory_static_max))
                     {
-                        // CA-359124 VM is migratable if a snapshot has more VIFs than the VM. As long as the mapping takes this into account. 
-                        if (failure.ErrorDescription.Count > 0 && failure.ErrorDescription[0] == Failure.VIF_NOT_IN_MAP &&
-                            SnapshotsContainExtraVIFs(vm))
-                            break;
-
-                        if (failure.ErrorDescription.Count > 0 && failure.ErrorDescription[0] == Failure.RBAC_PERMISSION_DENIED)
-                            failureReason = failure.Message.Split('\n')[0].TrimEnd('\r'); // we want the first line only
-                        else if (failure.ErrorDescription.Count > 1 && failure.ErrorDescription[1].Contains(Failure.DYNAMIC_MEMORY_CONTROL_UNAVAILABLE))
-                            failureReason = FriendlyErrorNames.DYNAMIC_MEMORY_CONTROL_UNAVAILABLE;
-                        // CA-378758: Ensure all hosts in pool hit `HOST_NOT_ENOUGH_FREE_MEMORY` before preventing migration

```

Comment:
I removed this because counting the hosts does not seem to be necessary any more after the order of checks was changed.

Reply:
yup, moving the `break` outside of the `try/catch` did it. much better now and it'll support any pool-wide setting going forward 🚀 nice job 
<img src='https://emojis.slackmojis.com/emojis/images/1643514065/210/yay.gif?1643514065' height='30' />